operator/cmd/mesh/uninstall.go

```diff
@@ -141,60 +133,39 @@ func uninstall(cmd *cobra.Command, ctx cli.Context, rootArgs *RootArgs, uiArgs *
 		}
 	}
 
-	opts := &helmreconciler.Options{DryRun: rootArgs.DryRun, Log: l, ProgressLog: progress.NewLog()}
-	var h *helmreconciler.HelmReconciler
-
+	pl := progress.NewLog()
 	// If the user is performing a purge install but also specified a revision or filename, we should warn
 	// that the purge will still remove all resources
 	if uiArgs.purge && (uiArgs.revision != "" || uiArgs.filename != "") {
 		l.LogAndPrint(PurgeWithRevisionOrOperatorSpecifiedWarning)
 	}
-	// If only revision flag is set, we would prune resources by the revision label.
-	// Otherwise we would merge the revision flag and the filename flag and delete resources by following the
-	// owning name label.
-	var iop *iopv1alpha1.IstioOperator
-	if uiArgs.filename == "" {
-		iop = &iopv1alpha1.IstioOperator{
-			Spec: &iopv1alpha1.IstioOperatorSpec{Profile: "empty", Revision: uiArgs.revision},
-		}
-	} else {
-		_, iop, err = manifest.GenManifests([]string{uiArgs.filename},
-			applyFlagAliases(uiArgs.set, uiArgs.manifestsPath, uiArgs.revision), uiArgs.force, nil, kubeClient, l)
-		if err != nil {
-			return err
-		}
-	}
 
-	h, err = helmreconciler.NewHelmReconciler(client, kubeClient, iop, opts)
-	if err != nil {
-		return fmt.Errorf("failed to create reconciler: %v", err)
-	}
-	objectsList, err := h.GetPrunedResources(uiArgs.revision, uiArgs.purge, "")
+	objectsList, err := uninstall2.GetPrunedResources(kubeClient, "", "", uiArgs.revision, uiArgs.purge)
 	if err != nil {
 		return err
 	}
-	preCheckWarnings(cmd, kubeClientWithRev, uiArgs, ctx.IstioNamespace(), uiArgs.revision, objectsList, nil, l, rootArgs.DryRun)
+	preCheckWarnings(cmd, kubeClient, uiArgs, ctx.IstioNamespace(), uiArgs.revision, objectsList, l, rootArgs.DryRun)
 
-	if err := h.DeleteObjectsList(objectsList); err != nil {
+	if err := uninstall2.DeleteObjectsList(kubeClient, rootArgs.DryRun, l, objectsList); err != nil {
```

Comment:
I'm assuming we still are not following Helm's uninstall order here?

(that's current state AFAIK, but I am making sure my understanding is still true with this PR)

Reply:
Yep, no intended change in that regard at this point